vm-linuxagent-upgrade/azuredeploy.json

```diff
@@ -0,0 +1,197 @@
+{
+    "$schema": "http://schema.management.azure.com/schemas/2014-04-01-preview/deploymentTemplate.json",
+    "contentVersion": "1.0.0.0",
+    "parameters" : {
+        "newStorageAccountName": {
+            "type": "string",
+            "defaultValue" : "uniquestorageaccountName"
```

Comment:
Please add metadata description for your parameters. Also please remove default values for resources that require globally unique names such as storage account, dns